app/components/competency-title-editor.js

```diff
@@ -42,19 +42,18 @@ export default Component.extend(Validations, ValidationErrorDisplay, {
 
   save: task(function* () {
     this.send('addErrorDisplayFor', 'title');
-    let {validations} = yield this.validate();
-    return new Promise((resolve, reject) => {
-      if (validations.get('isInvalid')) {
-        reject();
-      } else {
-        const competency = this.competency;
-        const title = this.title;
-        if (isPresent(competency)) {
-          competency.set('title', title);
-        }
-        this.send('clearErrorDisplay');
-        resolve();
+    const { validations } = yield this.validate();
+
+    if (validations.isValid) {
+      const { competency, title } = this.getProperties('competency', 'title');
+
+      if (isPresent(competency)) {
+        competency.set('title', title);
       }
-    });
+
+      this.send('clearErrorDisplay');
+    } else {
+      yield reject();
```

Comment:
I feel like this should be `return reject()` and not `yield`.

Reply:
I think I tried to return it but the promise just hung and the spinner wouldn't resolve.

Reply:
Ok, go ahead and revert this one then back to the old format. There is probably something else wrong with it and I'll take a look at some future point.

Reply:
Checked again. It must have been with `await` instead. `return reject()` works under `task`.